openverse_catalog/utilities/dag_doc_gen/dag_doc_generation.py

```diff
@@ -0,0 +1,228 @@
+"""
+This script generates a markdown documentation file which aggregates various pieces
+of information about all of our DAGs. The generated document has two sections: "DAGs
+by type" and "individual DAG documentation". Both sections have a small table of
+contents.
+
+The DAGs-by-type section shows DAG ID and schedule interval for all DAGS, and also shows
+dated & media type info for provider DAGs. Where possible, the DAG IDs link to
+individual documentation sections further in the document.
+
+The individual DAG documentation section pulls the DAG's `doc_md` section and renders
+it within the document.
+"""
+import logging
+from collections import defaultdict
+from pathlib import Path
+from typing import NamedTuple
+
+from airflow.models import DAG, DagBag
+from providers.provider_workflows import PROVIDER_WORKFLOWS, ProviderWorkflow
+
+
+log = logging.getLogger(__name__)
+# Silence noisy modules
+logging.getLogger("common.storage.media").setLevel(logging.WARNING)
+
+# Constants
+DAG_MD_PATH = Path(__file__).parent / "DAGs.md"
+DAG_FOLDER = Path(__file__).parents[2] / "dags"
+PREAMBLE = """\
+# DAGs
+
+_Note: this document is auto-generated and should not be manually edited_
+
+This document describes the DAGs available along with pertinent DAG information and
+the DAG's documentation.
+
+The DAGs are shown in two forms:
+
+ - [DAGs by Type](#dags-by-type)
+ - [Individual DAG documentation](#dag-documentation)
+
+# DAGs by Type
+
+The following are DAGs grouped by their primary tag.
+
+"""
+MIDAMBLE = """
```

Comment:
Same!

Reply:
Okay I legit just made this word up because it seemed appropriate and I didn't know what else to call this but I guess I should not be surprised it's a _real_ word 😂